idp-service/src/main/java/io/mosip/idp/services/ClientManagementServiceImpl.java

```diff
@@ -5,25 +5,117 @@
  */
 package io.mosip.idp.services;
 
-import io.mosip.idp.core.dto.ClientDetailRequest;
+import com.nimbusds.jose.jwk.JWK;
+import io.mosip.idp.core.dto.ClientDetailCreateRequest;
 import io.mosip.idp.core.dto.ClientDetailResponse;
+import io.mosip.idp.core.dto.ClientDetailUpdateRequest;
+import io.mosip.idp.core.exception.IdPException;
 import io.mosip.idp.core.spi.ClientManagementService;
+import io.mosip.idp.core.util.ErrorConstants;
+import io.mosip.idp.domain.ClientDetail;
+import io.mosip.idp.repositories.ClientDetailRepository;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Map;
+
 @Service
 public class ClientManagementServiceImpl implements ClientManagementService {
+    @Autowired
+    ClientDetailRepository clientDetailRepository;
 
     private static final Logger logger = LoggerFactory.getLogger(ClientManagementServiceImpl.class);
 
     @Override
-    public ClientDetailResponse createOIDCClient(ClientDetailRequest clientDetailRequest) {
-        return null;
+    public ClientDetailResponse createOIDCClient(ClientDetailCreateRequest clientDetailCreateRequest) throws IdPException {
+        var clientDetailFromDb = clientDetailRepository.findById(clientDetailCreateRequest.getClientId());
+
+        if (clientDetailFromDb.isPresent()) {
+            logger.error(ErrorConstants.DUPLICATE_CLIENT_ID);
+            throw new IdPException(ErrorConstants.DUPLICATE_CLIENT_ID);
+        }
+
+        var publicKeyJson = getJwkJson(clientDetailCreateRequest.getPublicKey());
+
+        if (publicKeyJson == null) {
```

Comment:
Instead of NULL check here, we can throw IdpException from getJwkJson() method @anshulv1401

Reply:
done